R/auth.R

```diff
@@ -0,0 +1,24 @@
+#' @title Get the API key
+#' @description Get the API key from the environment variable DELPHI_EPIDATA_KEY or getOption("delphi.epidata.key")
+#' @param verbose Will print the source of the key if TRUE (e.g. options, environment, or config file)
+#' @return The API key
+#' @export
+get_auth_key <- function() {
+  key <- Sys.getenv("DELPHI_EPIDATA_KEY", unset = "")
+  if (key != "") return(key)
+
+  key <- getOption("delphi.epidata.key", default = "")
+  if (key != "") return(key)
+
+  warning(
```

Comment:
suggestion: use `rlang::warn` with `.frequency`, `.frequency_id` options to have a cooldown on the warning.

Reply:
note: `rlang::warn` has a built-in `cli`-like format which might try to combine & wrap your lines.  (Or maybe that's only if you use actual `cli` via `use_cli_format=TRUE`.)  It might require a little adjustment to properly use its bulleted lists, etc.  But I think it'll enable better-looking messages and have suggested using `cli` below.